model/messages/consensus.go

```diff
@@ -17,12 +19,27 @@ func NewUntrustedProposal(internal *flow.Proposal) *UntrustedProposal {
 
 // DeclareTrusted converts the UntrustedProposal to a trusted internal flow.Proposal.
 // CAUTION: Prior to using this function, ensure that the untrusted proposal has been fully validated.
-// TODO(malleability immutable): This conversion should eventually be accompanied by a full validation of the untrusted input.
-func (msg *UntrustedProposal) DeclareTrusted() *flow.Proposal {
+//
+// All errors indicate that the input message could not be converted to a valid proposal.
+func (msg *UntrustedProposal) DeclareTrusted() (*flow.Proposal, error) {
```

Comment:
Could you please refactor this so that we apply mutation protection to `Proposal` as well?
- Add the `structwrite` directive to `Proposal`
- Add a `NewProposal` function which performs necessary checks (similar to what you have here)
- Change this function to simply convert types and call `NewProposal`